src/config/CodeGeneratorConfig.php

```diff
@@ -4,9 +4,9 @@
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
 class CodeGeneratorConfig {
-	
+
 	protected $options;
-	
+
 	public function __construct(array $options = []) {
```

Comment:
I just realized, the architecture is already here. The `OptionsResolver` component supports this. See here: http://symfony.com/doc/current/components/options_resolver.html#default-values-that-depend-on-another-option

Should be doable in `CodeFileGeneratorConfig::getDefaultOptions()`, could you check this for your case?